src/utils/cpg.ts

```diff
@@ -74,64 +75,107 @@ export class CPGExporter {
     this.criteria = criteria;
     this.nestedBranch = [];
     this.activityDefinitions = {};
-    this.libraries = this.createLibraries();
+    this.pathways = pathways;
   }
 
   export(): Bundle {
-    const library = this.libraries[0];
-    const cpgRecommendation = this.createPlanDefinition(
+    const strategyDefinition = this.createPlanDefinition(
       uuidv4(),
-      this.pathway.name,
-      this.pathway.description,
-      'recommendation',
-      library.id
+      `Strategy${this.pathway.name}`,
+      `Strategy for ${this.pathway.name}`,
+      'strategy'
     );
 
-    /*
+    this._export(strategyDefinition, this.pathway);
+    this.bundle.entry.push(createBundleEntry(strategyDefinition));
+    return this.bundle;
+  }
+
+  _export(
+    strategyDefinition: PlanDefinition,
+    pathway: Pathway,
+    referencedFrom?: PlanDefinitionAction,
+    condition?: PlanDefinitionCondition[]
+  ): void {
+    // the action for the strategy definition
+    const pathwayAction = createAction(pathway.name, `Action for ${pathway.name}`);
+
+    if (referencedFrom && referencedFrom.id && pathwayAction.id) {
```

Comment:
This can be consolidated. Also I think `pathwayAction.id` will always be set right? Is this just an issue with the typing being optional?
```suggestion
    if (referencedFrom?.id && pathwayAction.id) {
```

Reply:
yes id is optional

Reply:
Should it be? Is there ever a point in the builder where it isn't set? The type can be updated in `fhir-objects.d.ts`